sentinel-cluster/sentinel-cluster-client-redis/src/main/java/com/alibaba/csp/sentinel/cluster/redis/config/RedisFlowRuleManager.java

```diff
@@ -0,0 +1,137 @@
+package com.alibaba.csp.sentinel.cluster.redis.config;
+
+import com.alibaba.csp.sentinel.cluster.redis.RedisClient;
+import com.alibaba.csp.sentinel.cluster.redis.RedisClientFactory;
+import com.alibaba.csp.sentinel.log.RecordLog;
+import com.alibaba.csp.sentinel.property.DynamicSentinelProperty;
+import com.alibaba.csp.sentinel.property.PropertyListener;
+import com.alibaba.csp.sentinel.property.SentinelProperty;
+import com.alibaba.csp.sentinel.slots.block.flow.ClusterFlowConfig;
+import com.alibaba.csp.sentinel.slots.block.flow.FlowRule;
+import com.alibaba.csp.sentinel.slots.block.flow.FlowRuleUtil;
+import com.alibaba.csp.sentinel.util.AssertUtil;
+
+import java.util.*;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class RedisFlowRuleManager {
+    public static volatile boolean publishRuleToRedis = true;
+
+    private static Map<String, List<FlowRule>> flowRules = new HashMap<String, List<FlowRule>>();
+    private static Map<Long, FlowRule> flowIdToRule = new HashMap();
+    private static final RedisFlowPropertyListener LISTENER = new RedisFlowPropertyListener();
+    private static SentinelProperty<List<FlowRule>> currentProperty = new DynamicSentinelProperty<List<FlowRule>>();
+
+    static {
+        currentProperty.addListener(LISTENER);
+    }
+
+    public static void register2Property(SentinelProperty<List<FlowRule>> property) {
+        AssertUtil.notNull(property, "property cannot be null");
+        synchronized (LISTENER) {
+            RecordLog.info("[FlowRuleManager] Registering new property to flow rule manager");
+            currentProperty.removeListener(LISTENER);
+            property.addListener(LISTENER);
+            currentProperty = property;
+        }
+    }
+
+    public static void loadRules(List<FlowRule> rules) {
+        currentProperty.updateValue(rules);
+    }
+
+    private static void updateRule(List<FlowRule> confRules) {
+        RedisClientFactory factory = RedisClientFactoryManager.getFactory();
+        if (factory == null) {
+            RecordLog.warn(
+                    "[RedisFlowRuleManager]  cannot get RedisClientFactory, please init redis config");
+            return;
+        }
+
+        RedisClient redisClient = factory.getClient();
+
+        confRules = confRules == null ? new ArrayList<FlowRule>() : confRules;
+        Map<Long, FlowRule> oldRules = RedisFlowRuleManager.flowIdToRule;
+        updateLocaleRule(confRules);
+        resetRedisRuleAndMetrics(oldRules, redisClient);
+        clearInvalidRuleAndMetrics(oldRules, redisClient);
+
+        redisClient.close();
+    }
+
+    private static void updateLocaleRule(List<FlowRule> confRule) {
+        RedisFlowRuleManager.flowIdToRule = new ConcurrentHashMap<>();
```

Comment:
Here we should reduce the "gap". We should guarantee that the variable is always ready.